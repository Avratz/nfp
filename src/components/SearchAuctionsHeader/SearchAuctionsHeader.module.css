.header {
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
  padding-top: 4rem;
}

.subNavigation {
  margin: 0.5rem 0;
}

.headerFilters {
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
}
.divider {
  display: none;
}
@media screen and (min-width: 768px) {
  .headerFilters {
    flex-direction: row;
    justify-content: space-between;
    gap: 2rem;
    margin: 3rem 0;
  }
  .selectBox {
    width: 40%;
  }
  .subNavigation {
    width: 60%;
    justify-content: flex-end;
  }
  .searchIcon {
    background-color: var(--accent);
    border-radius: 50%;
    width: 50px;
    height: 50px;
    color: var(--neutral-8);
    right: 0;
  }
  .inputSearch {
    padding: 1rem 0;
    font-size: var(--text-2xl);
    font-weight: var(--font-regular);
  }
  .divider {
    display: block;
    margin-top: 0.5rem;
  }
}
