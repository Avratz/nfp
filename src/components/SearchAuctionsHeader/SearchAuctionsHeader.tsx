import { Input } from '@/ui/Input/Input'
import { SelectBox } from '@/ui/SelectBox/SelectBox'
import { ToggleButton } from '@/ui/ToggleButton/ToggleButton'
import { ToggleButtonGroup } from '@/ui/ToggleButtonGroup/ToggleButtonGroup'

import styles from './SearchAuctionsHeader.module.css'

export function SearchAuctionsHeader() {
  return (
    <header className={styles.header}>
      <Input placeholder="Search" icon={{ slot: 'right', name: 'search' }} />
      <SelectBox placeholder="Newest" setSelected={(value) => null}>
        Newest Oldest
      </SelectBox>

      <ToggleButtonGroup
        value="all-items"
        onChange={(value) => null}
        className={styles.subNavigation}
      >
        <ToggleButton value="all-items">All Items</ToggleButton>
        <ToggleButton value="art">Art</ToggleButton>
        <ToggleButton value="photography">Photography</ToggleButton>
      </ToggleButtonGroup>
    </header>
  )
}
