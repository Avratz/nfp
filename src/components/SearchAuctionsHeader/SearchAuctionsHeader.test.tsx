import { render } from '@testing-library/react'
import { SearchAuctionsHeader } from './SearchAuctionsHeader'

jest.mock('@/ui/Icon/Icon', () => {
  function Icon() {
    return <div></div>
  }

  Icon.displayName = 'Icon'
  return {
    __esModule: true,
    Icon,
  }
})
describe('SearchAuctionsHeader', () => {
  it('renders SearchAuctionsHeader', () => {
    let { container } = render(<SearchAuctionsHeader />)
    expect(container).toMatchSnapshot()
  })
})
