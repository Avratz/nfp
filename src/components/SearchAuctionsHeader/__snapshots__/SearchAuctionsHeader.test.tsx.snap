// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchAuctionsHeader renders SearchAuctionsHeader 1`] = `
<div>
  <header
    class="header"
  >
    <label
      class="inputLabel"
      for=":r0:"
    >
      <div
        class="inputWrapper"
      >
        <input
          class="input base inputRight inputSearch"
          id=":r0:"
          placeholder="Type to find something nice..."
          type="text"
        />
        <div />
      </div>
    </label>
    <hr
      class="divider horizontal divider"
    />
    <div
      class="headerFilters"
    >
      <div
        class="selectBox selectBox"
      >
        <button
          aria-expanded="false"
          aria-haspopup="listbox"
          class="button"
          data-headlessui-state=""
          id="headlessui-listbox-button-:r1:"
          type="button"
        >
          <span>
            Newest
          </span>
          <span
            class="icon"
          >
            <div />
          </span>
        </button>
      </div>
      <div
        class="toggleButtonGroup subNavigation"
        id="headlessui-radiogroup-:r3:"
        role="radiogroup"
      >
        <div
          aria-checked="true"
          class="toggleButton checked"
          data-headlessui-state="checked"
          id="headlessui-radiogroup-option-:r4:"
          role="radio"
          tabindex="0"
        >
          All Items
        </div>
        <div
          aria-checked="false"
          class="toggleButton"
          data-headlessui-state=""
          id="headlessui-radiogroup-option-:r5:"
          role="radio"
          tabindex="-1"
        >
          Art
        </div>
        <div
          aria-checked="false"
          class="toggleButton"
          data-headlessui-state=""
          id="headlessui-radiogroup-option-:r6:"
          role="radio"
          tabindex="-1"
        >
          Photography
        </div>
      </div>
    </div>
  </header>
</div>
`;
